Functions/Quantification/xASL_quant_FSL.m

```diff
@@ -0,0 +1,573 @@
+	%     %fprintf(option_file, '--sliceband=%i\n', sliceband);
+	%     %fprintf('BASIL: Multi-band setup with number of slices per band: %i\n', slicedband);
+	%
+	% 	%fprintf(option_file, '--t1im=%s\n', t1im)
+	%     %fprintf('BASIL: Using supplied T1 (tissue) image in BASIL: %s\n', $t1im)
+
+	%     %% Noise specification
+	%     % For small numbers of time points we need an informative noise prior.
+	%     % The user can specify an assumed SNR for this, or give prior estimated noise standard deviation below.
+	%     if ~isfield(x.Q,'BasilSNR') || ~x.Q.BasilSNR
+	%         x.Q.BasilSNR = 10;
+	%     end
+	%
+	%     if size(PWI, 4) < 5
+	%         x.Q.BasilNoisePrior = 1;
+	%         fprintf('BASIL: Small number of volumes (%i < 5): informative noise prior will be used\n', size(PWI, 4));
+	%     end
+	%     if isfield(x.Q,'BasilNoisePrior') && x.Q.BasilNoisePrior
+	%         % Use an informative noise prior
+	%         if ~isfield(x.Q,'BasilNoiseSD') || ~x.Q.BasilNoiseSD
+	%             fprintf('BASIL: Using SNR of %f to set noise std dev\n', x.Q.BasilSNR);
+	%             % Estimate signal magntiude FIXME brain mask assume half of voxels
+	%             mag_max = max(PWI, [], 4);
+	%             brain_mag = 2*xASL_stat_MeanNan(mag_max(:));
+	%             fprintf('BASIL: Mean maximum signal across brain: %f\n', brain_mag);
+	%             % This will correspond to whole brain CBF (roughly) - about 0.5 of GM
+	%             x.Q.BasilNoiseSD = sqrt(brain_mag * 2 / x.Q.BasilSNR);
+	%         end
+	%         fprintf('BASIL: Using a prior noise std.dev. of: %f\n', x.Q.BasilNoiseSD);
+	%         fprintf(FIDoptionFile, '--prior-noise-stddev=%f\n', x.Q.BasilNoiseSD);
+	%     end
+
+
+	%% 5. Extra BASIL fitting options
+	if x.Q.BASIL.bSpatial
+		fprintf('BASIL: Use automated spatial smoothing\n');
+		BasilOptions = [BasilOptions ' --spatial'];
+	end
+
+	if x.Q.BASIL.bInferT1
+		fprintf('BASIL: Infer variable T1 values\n');
+		BasilOptions = [BasilOptions ' --infert1'];
+	end
+
+	if x.Q.BASIL.bInferArt
+		fprintf('BASIL: Infer arterial BV and arrival time\n');
+		BasilOptions = [BasilOptions ' --inferart'];
+	end
+
+	switch (x.Q.BASIL.Exch)
+		case 'simple'
+			fprintf('BASIL Exchange model: Simple single compartment with T1 of blood, per white paper\n');
+			BasilOptions = [BasilOptions ' --exch=simple'];
+		case 'mix'
+			fprintf('BASIL Exchange model: Well-mixed\n');
+			BasilOptions = [BasilOptions ' --exch=mix'];
+		case '2cpt'
+			fprintf('BASIL Exchange model: A two compartment exchange model following Parkes & Tofts\n');
+			BasilOptions = [BasilOptions ' --exch=2cpt'];
+		case 'spa'
+			fprintf('BASIL Exchange model: A single pass approximation from St. Lawrence\n');
+			BasilOptions = [BasilOptions ' --exch=spa'];
+		otherwise
+			warning(['BASIL Exchange model: ' x.Q.BASIL.Exch ' not recognized.'])
+	end
+
+	switch (x.Q.BASIL.Disp)
+		case 'none'
+			fprintf('BASIL Dispersion model: none\n');
```

Comment:
Default option

Reply:
I suggest we put the default options there anyway, to make it clear for the user

Reply:
Like in the other comments - defaults are set and printed on screen. See if you agree with that...